generators/entity-client/templates/angular/src/main/webapp/app/entities/update/entity-management-update.component.spec.ts.ejs

```diff
@@ -26,64 +30,183 @@ import { HttpResponse } from '@angular/common/http';
 import { HttpClientTestingModule } from '@angular/common/http/testing';
 import { FormBuilder } from '@angular/forms';
 import { ActivatedRoute } from '@angular/router';
-import { of } from 'rxjs';
+import { of, Subject } from 'rxjs';
 
 import { <%= entityAngularName %>Service } from '../service/<%= entityFileName %>.service';
-import { <%= entityAngularName %> } from '../<%= entityFileName %>.model';
-<%_
-Object.keys(differentRelationships).forEach(key => {
-    if (differentRelationships[key].some(rel => rel.reference.owned)) {
-        const uniqueRel = differentRelationships[key][0];
-        if (uniqueRel.otherEntityAngularName !== entityAngularName) {
-            if (isBuiltInUser(uniqueRel.otherEntityAngularName)) {
-_%>
-import { User } from 'app/entities/user/user.model';
-<%_
-            } else {
-_%>
-import { <%= uniqueRel.otherEntityAngularName %> } from 'app/entities/<%= uniqueRel.otherEntityPath %>/<%= uniqueRel.otherEntityFileName %>.model';
-<%_
-            }
-        }
-    }
-});
-_%>
+import { I<%= entityAngularName %>, <%= entityAngularName %> } from '../<%= entityFileName %>.model';
+<%_ for (const relationshipsByEntityNeedingOptions of allRelationshipsByEntityNeedingOptions.filter(relationships => relationships[0].otherEntity.entityAngularName !== entityAngularName)) { _%>
+    <%_ const otherEntity = relationshipsByEntityNeedingOptions[0].otherEntity; _%>
+    <%_ if (isBuiltInUser(otherEntity.entityAngularName)) { _%>
+
+import { IUser, User } from 'app/entities/user/user.model';
+import { UserService } from 'app/entities/user/user.service';
+    <%_ } else { _%>
+import { I<%= otherEntity.entityAngularName %>, <%= otherEntity.entityAngularName %> } from 'app/entities/<%= otherEntity.entityFolderName %>/<%= otherEntity.entityFileName %>.model';
+import { <%= otherEntity.entityAngularName %>Service } from 'app/entities/<%= otherEntity.entityFolderName %>/service/<%= otherEntity.entityFileName %>.service';
+    <%_ } _%>
+<%_ } _%>
 
 import { <%= entityAngularName %>UpdateComponent } from './<%= entityFileName %>-update.component';
 
 describe('Component Tests', () => {
     describe('<%= entityAngularName %> Management Update Component', () => {
+        const activatedRouteData = new Subject();
+        const activatedRouteStub: any = {
+            data: activatedRouteData.asObservable(),
+        };
+
```

Comment:
```suggestion
        let activatedRouteData: Observable<I<%= entityAngularName %>>;
```
I think it's more natural to construct desired `ActivatedRoute` and then call `comp.ngOnInit();` as this is happening in real workflow.